### DISCUSSION TITLE
Future of P5LIVE

### DISCUSSION THREAD

**Discussion by julesvirallinen** [ffd8/P5LIVE#86] [General]
Hi @ffd8! 

P5LIVE is amazing and is still my daily driver for visuals even after coding a similar app in react. 

I'm wondering what your future for the app looks like. 

My version is stongly amended with more features, but I keep hitting walls with the structure of the app being a single big js-file. 

I see a couple options, and am willing to help develop this into something that can be developed more easily. 

The first is to slowly add linters, maybe typescript and split the logic into modules with types, so it becomes more of a real software project. 

The second would be to port it into a proper local app like electron (while still cleaning it up), so things like having a filesystem would be easier, and sketches could even be coded in an IDE and synced automatically. 

If you are up for it, I would gladly start with cleaning this up and trying to get the code to be safer, tested and to last time better, so this could be a tool that lasts time, and enables easier development in the future. 

Thoughts? 

Splitting code and adding types with TS would be a good first step in any possibility though, as that makes future maintenance a million times easier

**Comment by ffd8** [comment]
Hey @julesvirallinen – apologies for the lag in addressing this – work + kiddo + projects + research + ______ brings this labor of love in and out of focus. i appreciate your intentions and interest, however i politely decline the suggested offer for a few reasons: P5LIVE has been a huuuge learning project for me over the years, with each added feature causing a deep tech dive and gaining coding skills along the way = my code is messy AF! sadly i never made a dev-blog, since i could have shared soo much trial+error insights from each feature it has. i also purposely don't follow a single style (linting faaaail;) as i've developed as a coder along the way, changed habits/styles, etc – which to me is important that this dev process leave those traces behind. i'm not a fan/interested in the TS way - plain ol vanilla JS is enough and this crazy single big file was on purpose; initially one could run it easily offline by just turning the dir into a live-server (used to suggest python until MacOS stopped shipping with it) + I'm not a fan of bundlers or super split up code. obviously this is not my training and recent experience which turned a 1-pager project into tons of components, gave me an appreciation for how we all code differently and should embrace that (unless it's a huge project with lots of hands on deck). it surely looks like chaos from the outside, which i like to call `duct tape programming`, but it makes total sense for me and is easily maintainable. many more features are planned, and i'll be keeping P5LIVE [a]live as software/hardware change in the future.

**Reply by julesvirallinen** [reply]
thanks @ffd8 for the response

I guess half of the question was: 

- how do you feel about unified code formatting / linting / any sort of work to make it possible for any sort of collaborative development

reason I'm asking is that P5LIVE is really good, and could have a lot of potential as something many in the community could help make even better. As you said it's clear from the code that it wasn't made by a developer, but the end project works amazingly well. I even coded a P5LIVE-type live editor from scratch at some point, but never really got away from my P5LIVE frankenstein-fork-monster. 

"it makes total sense for me and is easily maintainable" I totally get that -  if you want this to stay so that only you can develop it, that's your choice! 

I'm also glad to see you're considering developing P5LIVE further, when I made this thread for discussion it had been around 2 years since the previous breath of life in development, which was worrying. I'm asking in part because I can't be the only professional developer in this world passionate about p5.js coding, and willing to help :)
